h/links.py

```diff
@@ -25,7 +26,12 @@ def pretty_link(url):
 
 
 def html_link(request, annotation):
-    """Generate a link to an HTML representation of an annotation."""
+    """Return a link to an HTML representation of the given annotation, or None."""
+    is_third_party_annotation = annotation.authority != auth.authority(request)
```

Comment:
Elsewhere in the app we mostly seem to prefer `request.authority` rather than `auth.authority(request)`, which are the same if I understand the set up in `h.auth.__init__` correctly.

Reply:
Yep. On a quick grep I see about 22 uses of `request.authority`, vs only one use of `auth.authority`, on master

Reply:
And `request.authority` is just a shortcut for `auth.authority`. Ok, sounds good

Reply:
Fairly big simplification. Also note that one of my tests had a param that it wasn't using which a linter would have caught if we used linters properly for h.

```diff
diff --git a/h/links.py b/h/links.py
index 2fe2e6d35..fe6f502ff 100644
--- a/h/links.py
+++ b/h/links.py
@@ -5,7 +5,6 @@ Provides links to different representations of annotations.
 """
 
 
-from h import auth
 from h._compat import urlparse, url_unquote
 
 
@@ -27,7 +26,7 @@ def pretty_link(url):
 
 def html_link(request, annotation):
     """Return a link to an HTML representation of the given annotation, or None."""
-    is_third_party_annotation = annotation.authority != auth.authority(request)
+    is_third_party_annotation = annotation.authority != request.authority
     if is_third_party_annotation:
         # We don't currently support HTML representations of third party
         # annotations.
diff --git a/tests/h/links_test.py b/tests/h/links_test.py
index af2040258..b136f63e0 100644
--- a/tests/h/links_test.py
+++ b/tests/h/links_test.py
@@ -25,27 +25,20 @@ class FakeDocument(object):
         self.document_uris = []
 
 
-@pytest.mark.usefixtures('auth', 'routes')
+@pytest.mark.usefixtures('routes')
 class TestHTMLLink(object):
 
-    def test_html_link_looks_up_the_first_party_authority(
-            self, annotation, auth, pyramid_request):
-        """It should fetch the first-party authority of the h instance."""
-        links.html_link(pyramid_request, annotation)
-
-        auth.authority.assert_called_once_with(pyramid_request)
-
     def test_html_link_returns_links_for_first_party_annotations(
-            self, annotation, auth, pyramid_request):
+            self, annotation, pyramid_request):
         # Specify the authority so that it's a first-party annotation.
-        annotation.authority = auth.authority.return_value
+        annotation.authority = pyramid_request.authority
 
         link = links.html_link(pyramid_request, annotation)
 
         assert link == 'http://example.com/a/ANNOTATION_ID'
 
     def test_html_link_returns_None_for_third_party_annotations(
-            self, annotation, auth, pyramid_request):
+            self, annotation, pyramid_request):
         # Specify the authority so that it's a third-party annotation.
         annotation.authority = 'elifesciences.org'
 
@@ -60,10 +53,6 @@ class TestHTMLLink(object):
             id='ANNOTATION_ID',
         )
 
-    @pytest.fixture
-    def auth(self, patch):
-        return patch('h.links.auth')
-
     @pytest.fixture
     def routes(self, pyramid_config):
         pyramid_config.add_route('annotation', '/a/{id}')
```